File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.py
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        ans = 0
        left = 0
        m = collections.Counter()
        for right in range(len(nums)):
            m[nums[right]] += 1
            while m[nums[right]] > k:
                m[nums[left]] -= 1
                left += 1
            ans = max(ans,right - left + 1)
        
        return ans

File: 3225-length-of-longest-subarray-with-at-most-k-frequency/length-of-longest-subarray-with-at-most-k-frequency.cpp
class Solution {
public:
    int maxSubarrayLength(vector<int>& nums, int k) {
        int n = nums.size();
        int ans = 0;
        unordered_map<int,int> m;
        int l = 0;

        for(int i = 0 ; i < n ; i++){
            m[nums[i]]++;

            while ( m[nums[i]] > k){
                m[nums[l]]--;
                l++;
            }

            ans = max(ans,i-l+1);
        }

        return ans;
    }
};